packages/nimble-components/src/rich-text-editor/models/bold-button.ts

```diff
@@ -0,0 +1,33 @@
+import { html } from '@microsoft/fast-element';
+import type { Editor } from '@tiptap/core';
+import { EditorButton } from './editor-button';
+import { iconBoldBTag } from '../../icons/bold-b';
+
+/**
+ * Derived class from EditorButton for handling bold button events.
+ */
+export class BoldButton extends EditorButton {
+    public constructor(public override tiptapEditor: Editor) {
+        super(tiptapEditor);
+        this.class = 'bold';
+        this.iconLabel = 'Bold';
+        this.tiptapNodeOrMarkName = 'bold';
+        this.iconTemplate = html`<${iconBoldBTag} slot="start"></${iconBoldBTag}>`;
```

Comment:
I have updated it to have a single initialization in the base class and passed the specific values from the derived class. We thought that the interface name can be `ButtonProperties` as it carries the properties of the buttons instead of `ButtonInitializers` and updated the same for the abstract method too.